api/v1beta1/placementapi_types.go

```diff
@@ -159,6 +164,14 @@ type PasswordSelector struct {
 	Service string `json:"service"`
 }
 
+// PlacementImmutable defines the observed state of PlacementImmutable
+type PlacementAPIImmutable struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=true
+	// Make databaseInstance immutable
+	DatabaseInstance bool `json:"databaseInstance"`
```

Comment:
This will not work well with webhooks and json ser/deserialization. The webhook ser/des step will default this to false (the golang default) when the field is not provided in the request. You need to make this a *bool to avoid this.

Reply:
Fixed in the latest version.